contracts/REXLimitOrderManager.sol

```diff
@@ -0,0 +1,172 @@
+        ISuperToken token = ISuperToken(market.inputToken());
+
+        require(
+            token.isOperatorFor(address(this), msg.sender),
+            "No ACL permission"
+        );
+
+        bytes32 taskId = createGelatoTask(msg.sender, _market);
+
+        limitOrders[msg.sender][_market] = LimitOrder(
+            _isInverted,
+            _streamRate,
+            _price,
+            taskId,
+            false
+        );
+        emit LimitOrderCreated(
+            msg.sender,
+            _market,
+            _isInverted,
+            _streamRate,
+            _price
+        );
+    }
+
+    /**
+     *
+     * @notice Cancels a limit order.
+     * @dev The order is cancelled for the specified market, msg.sender is used for user's address.
+     * @param _market The address of the REX market where the order is placed for.
+     */
+    function cancelLimitOrder(address _market) public {
+        LimitOrder memory order = limitOrders[msg.sender][_market];
+        order.executed = true; // fail safe
+
+        cancelGelatoTask(order.taskId);
+    }
+
+    function createGelatoTask(
+        address _user,
+        address _market
+    ) internal returns (bytes32 taskId) {
+        ModuleData memory moduleData = ModuleData({
+            modules: new Module[](2),
+            args: new bytes[](2)
+        });
+
+        moduleData.modules[0] = Module.RESOLVER;
+        moduleData.modules[1] = Module.SINGLE_EXEC;
+
+        moduleData.args[0] = _resolverModuleArg(
+            address(this),
+            abi.encodeCall(this.checker, (_user, _market))
+        );
+
+        moduleData.args[1] = _proxyModuleArg();
+
+        taskId = _createTask(
+            address(this),
+            abi.encode(this.updateUserStream, abi.encodePacked(_user, _market)),
+            moduleData,
+            ETH
+        );
+    }
+
+    function cancelGelatoTask(bytes32 _taskId) internal {
+        _cancelTask(_taskId);
+    }
+
+    /**
+     *
+     * @notice Updates the user's stream if order is in limit.
+     * @dev The user's stream is updated based on the current price of the market.
+     * @param _user The address of the user whose stream is updated.
+     * @param _market The address of the REX market where the order is placed for.
+     */
+    function updateUserStream(address _user, address _market) external {
+        LimitOrder memory order = limitOrders[_user][_market];
+        require(order.executed == false, "Already executed");
+
+        IREXUniswapV3Market market = IREXUniswapV3Market(_market);
+        ISuperToken token = ISuperToken(market.inputToken());
+        uint256 price = uint256(uint(market.getLatestPrice()));
+        if (price < order.price) {
+            token.createFlowFrom(_user, _market, order.streamRate);
+        }
+    }
+
+    function checker(
+        address _user,
+        address _market
+    ) external view returns (bool canExec, bytes memory execPayload) {
```

Comment:
Somewhere in here you want to do two things:
1) Check if the price is above or below the limit, update the stream accordingly, turn it on or off
2) Check if the limit order expired, cancel the limit order

The checker should toggle the stream on and off until the expiration, at expiration it should cancel the limit order.

Reply:
@mikeghen will the user have option to change the expiration time?

Reply:
I have added limit order expiration time and changed the logic a bit, if the limit order is not triggered in the specific time period ( it will cancel the order automatically ). But if triggered once and the time has passed (user will have to close the position manually)